ngraph/core/src/op/negative.cpp

```diff
@@ -10,7 +10,7 @@
 using namespace std;
 using namespace ngraph;
 
-constexpr NodeTypeInfo op::Negative::type_info;
+NGRAPH_RTTI_DEFINITION(op::v0::Negative, "Negative", 0, util::UnaryElementwiseArithmetic);
 
```

Comment:
The `evaluate` method is implemented for boolean type which I think is not correct, because the boolean type is unsigned. Please, remove.

The `validate_and_infer_type` should prohibit this type as well. I understand that this is a backward incompatible change, but I would consider this as a bug. @ilyachur , what do you think about this?

Reply:
@lazarevevgeny I agree with you.

Reply:
I added separate check for boolean type inside validate_and_infer_types, because I think that it may be more clear, but if You think that it is not necessary, then I will remove it.

Reply:
@lazarevevgeny I also updated the supported types in the spec. Is it ok?